tests/Support/SupportLazyCollectionIsLazyTest.php

```diff
@@ -986,6 +987,33 @@ public function testSliceIsLazy()
         });
     }
 
+    public function testFindFirstOrFailIsLazy()
+    {
+        $this->assertEnumerates(1, function ($collection) {
+            try {
+                $collection->firstOrFail();
+            } catch (ItemNotFoundException $e) {
+                //
+            }
+        });
+
+        $this->assertEnumerates(1, function ($collection) {
+            $collection->firstOrFail(function ($item) {
+                return $item === 1;
+            });
+        });
+
+        $this->assertEnumerates(2, function ($collection) {
+            try {
+                $collection->firstOrFail(function ($item) {
+                    return $item % 2 === 0;
+                });
+            } catch (ItemNotFoundException $e) {
+                //
+            }
```

Comment:
Same here. Why would this fail?

And really, you're not testing an actual failure, and ensuring it enumerates the whole collection once.